ensime_shared/ensime.py

```diff
@@ -278,12 +278,10 @@ def au_buf_leave(self, client, filename):
         client.buffer_leave(filename)
 
     @execute_with_client()
-    def au_cursor_hold(self, client, filename):
-        client.on_cursor_hold(filename)
-
-    @execute_with_client(quiet=True)
-    def au_cursor_moved(self, client, filename):
-        client.on_cursor_move(filename)
+    def refresh_messages(self, client):
+        filename = client.editor.path()
+        if filename.endswith(".scala"):
```

Comment:
There are a few lines of code in `fun_en_complete_func` that could be factored out into a function for checking "is the buffer one that ensime-vim supports?". It uses Vim's notion of `filetype`, which I think is a better way to go than file extension.